.github/workflows/ci.yml

```
@@ -0,0 +1,25 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
```

`18.x` という指定は公式に認められてないですね。

```suggestion
          node-version: '18'
```

あと、 `setup-node` は簡単にキャッシュが使えるので追加しておくと良いです。
https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md